Lab20.md

```diff
@@ -14,38 +15,125 @@ wget https://raw.githubusercontent.com/rook/rook/master/cluster/examples/kuberne
 wget https://raw.githubusercontent.com/rook/rook/master/cluster/examples/kubernetes/mysql.yaml
 ```
 
+### Wordpress Description
+
+Examine the Wordpress application descriptions. Take note of:
+
+* PersistentVolumeClaim (wp-pv-claim)
+* volumeMounts (wordpress-persistent-storage)
+```
+more wordpress.yaml
+```
+
+How much storage was allocated for Wordpress?
+What type of storage was allocated?
+Where in the container in the storage mounted?
+What does "accessModes: ReadWriteOnce" mean?
+
+### MySQL Description
+
+Examine the MySQL application descriptions. Take note of:
+
+* PersistentVolumeClaim (mysql-pv-claim)
+* volumeMounts (mysql-persistent-storage)
+
+```
+more mysql.yaml
+```
+
+How much storage was allocated for Wordpress?
+What type of storage was allocated?
+Why is this claim labeled with "app: wordpress"?
+
 
 ## Create Applications
 
+Go ahead and start up the applications which in turn will also allocate the storage through Rook.
 ```
 kubectl create -f mysql.yaml
 kubectl create -f wordpress.yaml
 ```
 
-Then let's wait for the pods to be ready
+## Volume Claims
+
+The PVCs (Persistent Volume Claims) will be allocated as part of this deployment. Watch as Rook allocates these storage resources.
+
 ```
-watch -n1 -d kubectl get pods
+watch -n1 -d kubectl get PersistentVolumeClaim
 ```
-ctrl-c once the pods are `Running`
+
+Once the storage is allocated, "ctrl-c" out of this watch and verify that the rest of the deployment has started.
 
 ## Examine the created resources
 
+Validate that all the Wordpress and MySQL pods are running.
+
+```
+watch -n1 -d kubectl get pods
+```
+
+Once all the pods are started, "ctrl-c" out of this watch and verify that the full deployment and services are running OK.
+
 ```
 kubectl get Deployments
-kubectl get PersistentVolumeClaim
 kubectl get Services
 ```
 
-Run `rbd ls -p replicapool` in the toolbox container
+## Examine Ceph setup via Toolbox
+
+Let's take a deeper look at has been setup in Ceph. Pull up the Ceph Toolbox container and run through some Ceph commands to examine the configuration.
+
+```
+kubectl apply -f https://raw.githubusercontent.com/rook/rook/release-1.0/cluster/examples/kubernetes/ceph/toolbox.yaml
```

Comment:
The Ceph Toolbox should already be created, so we the `kubectl exec` should be enough.

Maybe back in the toolbox lab we should ask them to keep it open in another window/tab since we use the toolbox a couple of times later?